core/shared/src/main/scala/zio/stm/ZSTM.scala

```diff
@@ -2051,6 +2066,18 @@ object ZSTM {
         _isChanged = self.isChanged
       }
 
+      /**
+       * Resets the Entry with a given value.
+       */
+      def reset(resetValue: Any): Entry = new Entry {
```

Comment:
This cannot be public.

Reply:
@jdegoes Thanks for the review! I have incorporated it and pushed the changes. Please review it, whenever you are available. Thanks!